pkg/workspace/library_execution.go

```diff
@@ -86,7 +87,34 @@ func (ll *LibraryExecution) Values(valuesOverlays []*datavalues.Envelope, schema
 		loader:         loader,
 	}
 
-	return dvpp.Apply()
+	values, libValues, err := dvpp.Apply()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return values, libValues, ll.validateValues(values, loader)
+}
+
+// validateValues runs validations from @assert/validate annotations in Data Values for the current library.
+//
+// Returns an error if the arguments to an @assert/validate are invalid,
+// otherwise, checks the assertionChecker for violations, and returns nil if there are no violations.
+func (ll *LibraryExecution) validateValues(values *datavalues.Envelope, loader *TemplateLoader) error {
```

Comment:
Keep an eye open for the possibility that we'll want to wrap the validation process in a struct of its own (like we do with `DataValuesSchemaPreProcessing`, `DataValuesPreProcessing`, and `OverlayPostProcessing`). 

Among other benefits is that it provides an easy home for helper functions and a handy place to put state/inject dependencies.

Could be a bit early, but wanted to note that existing pattern.